py_openda/py_openda/costFunctions/LorenzStochModelInstance.py

```diff
@@ -0,0 +1,166 @@
+Created on Thu Nov 22 11:32:08 2018
+
+@author: hegeman
+"""
+
+from math import sqrt
+import numpy as np
+from scipy.stats import norm
+from scipy.integrate import ode
+from py_openda.costFunctions.JObjects import PyTime
+import py_openda.utils.py4j_utils as utils
+
+class LorenzStochModelInstance:
+    """
+    Instance of a three point Lorenz model.
+    """
+    def __init__(self, model_attributes, noise_config, main_or_ens=None):
+        """
+        :param model_attributes: attributes saved in the model factory.
+        :param noise_config: dictionary as given by EnkfAlgorithm.xml for the noise configuration.
+        :param main_or_ens: determines the ouput level of the model.
+        """
+        (self.param, self.param_uncertainty, self.state, self.state_uncertainty, self.sys_mean,
+         self.sys_std, self.span) = model_attributes
+        
+        if noise_config is None:
+            if main_or_ens == "main":
+                noise_config = {'@stochParameter':False, '@stochForcing':False, '@stochInit':False}
+            elif main_or_ens == "ens":
+                noise_config = {'@stochParameter':False, '@stochForcing':True, '@stochInit':True}
+        if noise_config.get('@stochInit'):
+            realizations = [norm(loc=mean, scale=std).rvs() for mean,
+                            std in zip(self.state, self.state_uncertainty)]
+            self.state = realizations.copy()
+        if noise_config.get('@stochParameter'):
+            realizations = [norm(loc=mean, scale=std).rvs() for mean,
+                            std in zip(list(self.param.values()),
+                                       list(self.param_uncertainty.values()))]
+            self.param = realizations
+
+        self.auto_noise = noise_config.get('@stochForcing')
+
+
+        self.current_time = PyTime(self.span[0])
+        self.state = np.array(self.state)
+
+
+    def get_time_horizon(self):
+        """
+        Get the computational time horizon of the model (begin and end time).
+
+        :return: the time horizon (containing begin and end time).
+        """
+        return PyTime(self.span[0], self.span[2])
+
+    def get_current_time(self):
+        """
+        Get the model instance's current simulation time stamp.
+
+        :return: The model's current simulation time stamp.
+        """
+        return self.current_time
+
+    def announce_observed_values(self, descriptions):
+        """
+        Tells model that it can expect to be asked for model values corresponding to the observations
+        described. The model can make arrangement to save these values. The method compute run over a long
+        interval at once, not stopping at each time with observations.
+        This is meant to increase the performance especially of calibration algorithms.
+
+        :param descriptions: an ObservationDescriptions object with meta data for the observations.
+        :return:
+        """
+        None
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Statement seems to have no effect](https://app.codacy.com/gh/OpenDA-Association/OpenDA/pullRequest?prid=4902598)